test/Microsoft.AspNetCore.TestHost.Tests/TestServerTests.cs

```diff
@@ -71,6 +71,34 @@ public class ThirdPartyContainerStartup
                     $"{ctx.RequestServices.GetRequiredService<SimpleService>().Message}, {ctx.RequestServices.GetRequiredService<TestService>().Message}"));
         }
 
+        [Fact]
+        public async Task ExternalContainerInstanceCanBeUsedForEverything()
```

Comment:
How does this verify the custom container is being used?

Reply:
How would you like it to be verified? Can you think of all your comments and do them all at once?

Reply:
I am not that familiar with your testing framework, I used one of the tests as example and run it in  debugger to verify. Feel free to change it any way you like.

Reply:
Heh, sorry, I have a bad habit of re-reviewing things where they are big changes. A good test would be to add a service in the ExternalContainer implementation and make sure it's resolved in the Startup constructor.

I would also not add a test to TestServerTests.cs I would add a test to https://github.com/aspnet/Hosting/blob/dev/test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs

Reply:
Is this it? Think about it, I am catching flight in a few hours. I have time just for one more change.

Reply:
I can take over if you want. It's pretty much done.

Reply:
I would appreciate if you do. I am getting low on time.